doc/api_ref/ffi.rst

```diff
@@ -874,6 +874,100 @@ Object Identifiers
    Return 1 if ``a`` is equal to ``b``, 0 if ``a`` is not equal to ``b``
 
 
+EC Groups
+----------------------------------------
+
+.. versionadded:: 3.8.0
+
+.. cpp:type:: opaque* botan_ec_group_t
+
+   An opaque data type for an EC Group. Don't mess with it.
+
+.. cpp:function:: int botan_ec_group_destroy(botan_ec_group_t oid)
+
+   Destroy an object.
+
+.. cpp:function:: int botan_ec_group_supports_application_specific_group(int* out)
+
+   Checks if in this build configuration it is possible to register an application specific elliptic curve,
+   and sets ``out`` to 1 if so, 0 otherwise.
+
+.. cpp:function:: int botan_ec_group_supports_named_group(const char* name, int* out)
+
+   Checks if in this build configuration botan_ec_group_from_name(group_ptr, name) will succeed,
+   and sets ``out`` to 1 if so, 0 otherwise.
+
+.. cpp:function:: int botan_ec_group_from_params(botan_ec_group_t* ec_group, \
+                               botan_asn1_oid_t oid, \
+                               botan_mp_t p, \
+                               botan_mp_t a, \
+                               botan_mp_t b, \
+                               botan_mp_t base_x, \
+                               botan_mp_t base_y, \
+                               botan_mp_t order)
+
+   Create a new EC Group from the given parameters
```

Comment:
Maybe worth referencing here the docs of the relevant `EC_Group` constructor which explains the restrictions on sizes/etc

Reply:
Can I somehow make a reference directly to that constructor? I tried adding `.. _ec_group_param_constructor:` directly before it, but it complained about duplicate references. I also tried using the function signature directly, that didn't work either, but I'm also not very knowledgeable about rst

Reply:
Offhand if those don't work I'm not sure what Sphinx wants here. I'll mess with it later